### ISSUE TITLE
Problems on submitting prediction results

### ISSUE THREAD

**Issue by Xiangxu-0103** [waymo-research/waymo-open-dataset#484]
Hi, waymo teams.
     When I submit our preidction results on the challenge of 3d semantic seg server following the latest tutorial, it will report the error `INVALID_ARGUMENT: Failed to parse the prediction from the given file.`

Wish your help.

**Comment by charlesq34** [comment]
@Xiangxu-0103 Have you tried to follow the example of creating a dummy submission of https://github.com/waymo-research/waymo-open-dataset/blob/master/tutorial/tutorial_3d_semseg.ipynb ?
It could be that the file you generated is not in the right proto format.

**Comment by Xiangxu-0103** [comment]
The problem has been solved. When submitting the validation set results, we should write `segmentation_frame_list.SerializeToString()` into `bin` file instead of `submission.SerializeToString()` in the tutorial.

**Comment by jialeli1** [comment]
Hey guys, could you share the number of annotated training/val samples? Just for the lidar seg dataset decoding check usage.
@charlesq34 @Xiangxu-0103

**Comment by charlesq34** [comment]
@Xiangxu-0103 good to know it is resolved.
I believe the SegmentationFrameList is used for the segmentation metric tool (https://github.com/waymo-research/waymo-open-dataset/blob/master/waymo_open_dataset/metrics/tools/compute_segmentation_metrics_main.cc)
For the eval server submission (on the WebUI), one still needs to submit the file of the submission proto (https://github.com/waymo-research/waymo-open-dataset/blob/master/waymo_open_dataset/protos/segmentation_submission.proto)

**Comment by charlesq34** [comment]
@jialeli1 
for the most recent v1.3.2 version, numbers of frames with 3d semseg labels are:
Test set: 2982
Val set: 5976
Train set: 23691

**Comment by jialeli1** [comment]
Thanks for your kind answer and tips. :thumbsup:

**Comment by Xiangxu-0103** [comment]
Hi, @charlesq34 , sorry to bother you again.
I follow the [tutorial]( https://github.com/waymo-research/waymo-open-dataset/blob/master/tutorial/tutorial_3d_semseg.ipynb) to create a dummy submission for test set and submit it to eval server. The error `INVALID_ARGUMENT: Failed to parse the prediction from the given file.` occurs again.  😢

**Comment by charlesq34** [comment]
@Xiangxu-0103 I tried again, it worked for me. Did you change the email address (account_name)? It should match your account.

**Comment by Xiangxu-0103** [comment]
Yes, I have changed the email address to my own email.

**Comment by charlesq34** [comment]
You can try to load it back in the notebook and check its fields:
```
with gfile.Open(filename, 'rb') as fp:
  submission = segmentation_submission_pb2.SemanticSegmentationSubmission()
  submission.ParseFromString(fp.read())
print(submission)
```
where the `filename` is the file you used to submit to the eval server.

**Comment by Xiangxu-0103** [comment]
@charlesq34 , I print the submission info, and the following is what I get:
![print](https://user-images.githubusercontent.com/55445986/168504229-99e568ba-cf6c-4f2a-845c-7b15e206913f.png)

**Comment by charlesq34** [comment]
@Xiangxu-0103 
the print looks good to me.
One thing to note is that some users mis-configured the shape of segmentation_label_compressed.

The segmentation labels should be in shape [H, W, 2] where the second channel is the semseg results. Some provided submission has that in shape [H, W].
Ref: https://github.com/waymo-research/waymo-open-dataset/blob/520c34a2090460d7498078fa9033e011e28e86b0/waymo_open_dataset/dataset.proto#L260 

You can try to decompress the field and see its shape.

**Comment by Xiangxu-0103** [comment]
Thanks for your nicely response, and the problem has been solved successfully.
